src/Kovan-DssSpell.sol

```diff
@@ -16,39 +16,284 @@
+        // Init RWA-00x in Jug
+        Initializable(DssExecLib.jug()).init(collateral.ilk);
+
+        // Allow RWA-00x Join to modify Vat registry
+        DssExecLib.authorize(vat, collateral.MCD_JOIN);
+
+        // Allow RwaLiquidationOracle to modify Vat registry
+        // DssExecLib.authorize(vat, MIP21_LIQUIDATION_ORACLE);
+
+        // Increase the global debt ceiling by the ilk ceiling
+        DssExecLib.increaseGlobalDebtCeiling(collateral.CEIL);
+        // Set the ilk debt ceiling
+        DssExecLib.setIlkDebtCeiling(collateral.ilk, collateral.CEIL);
+
+        // No dust
+        // DssExecLib.setIlkMinVaultAmount(collateral.ilk, 0);
+
+        // stability fee
+        DssExecLib.setIlkStabilityFee(collateral.ilk, collateral.RATE, false);
+
+        // collateralization ratio
+        DssExecLib.setIlkLiquidationRatio(collateral.ilk, collateral.MAT);
+
+        // poke the spotter to pull in a price
+        DssExecLib.updateCollateralPrice(collateral.ilk);
+
+        // give the urn permissions on the join adapter
+        // DssExecLib.authorize(collateral.MCD_JOIN, collateral.URN);
+
+        // set up the urn
+        Hopeable(collateral.URN).hope(collateral.OPERATOR);
+
+        // set up output conduit
+        // Hopeable(collateral.OUTPUT_CONDUIT).hope(collateral.OPERATOR));
+
+        // Authorize the SC Domain team deployer address on the output conduit
+        // during introductory phase. This allows the SC team to assist in the
+        // testing of a complete circuit. Once a broker dealer arrangement is
+        // established the deployer address should be `deny`ed on the conduit.
+        // Kissable(collateral.OUTPUT_CONDUIT).kiss(SC_DOMAIN_DEPLOYER_07);
+
+        // add RWA-00x contract to the changelog
+        DssExecLib.setChangelogAddress(collateral.gemID, collateral.GEM);
+        DssExecLib.setChangelogAddress(collateral.pipID, pip);
+        DssExecLib.setChangelogAddress(collateral.joinID, collateral.MCD_JOIN);
+        DssExecLib.setChangelogAddress(collateral.urnID, collateral.URN);
+        DssExecLib.setChangelogAddress(
+            collateral.inputConduitID, collateral.INPUT_CONDUIT
+        );
+        DssExecLib.setChangelogAddress(
+            collateral.outputConduitID, collateral.OUTPUT_CONDUIT
+        );
+
+        address ILK_REGISTRY = DssExecLib.getChangelogAddress("ILK_REGISTRY");
+        IlkRegistryAbstract(ILK_REGISTRY).put(
+            collateral.ilk,
+            collateral.MCD_JOIN,
+            collateral.GEM,
+            DSTokenAbstract(collateral.GEM).decimals(),
+            3,
+            pip,
+            address(0),
+            collateral.ilkRegistryName,
+            bytes32ToStr(collateral.ilk)
+        );
+    }
+
+    function bytes32ToStr(bytes32 _bytes32) internal pure returns (string memory) {
+        bytes memory bytesArray = new bytes(32);
+        for (uint256 i; i < 32; i++) {
+            bytesArray[i] = _bytes32[i];
+        }
```

Comment:
Someone reported a bug with this function in #dev.  It seems as though it was not null terminating the string or something.  Let's investigate more.

Reply:
I just modified one of the spell tests to actually compare the ilk to a bytes32 version, and this indeed caught that they weren't matching exactly. To remedy this, I've simply added a string version of the ilk to the struct in the spell, which seems like the easiest fix for now.